WebApiBackend/WebApiBackend/Controllers/FlatController.cs

```diff
@@ -48,5 +49,47 @@ public ActionResult<List<DisplayMemberDTO>> GetMembers()
             return _MemberMapper.Map<List<DisplayMemberDTO>>(members);
         }
 
+        [Authorize]
+        [HttpGet("AddUserToFlat/{userName}")]    
+        public ActionResult<AddUserToFlatDto> AddUserToFlat( string userName)
+        {
+            var identity = (ClaimsIdentity)HttpContext.User.Identity;
+            int userID = Int16.Parse(identity.FindFirst(ClaimTypes.NameIdentifier).Value);
+            
+            //the operating user and his flat
+            var user = _context.User.Find(userID);
+            Flat flat = _context.Flat.Where(f => f.Id == user.FlatId).FirstOrDefault();
+
+            //the user who is operated
+            User finduser = _context.User.Where(u => u.UserName == userName).FirstOrDefault();
+
+            UserDTO ret = new UserDTO();
```

Comment:
Sorry just saw this. You need to change UserDTO so that it has an empty constructor. Right now its only constructor takes a user as its parameter.

Reply:
Changed quickly myself